### ISSUE TITLE
Provide a "writable" D-Bus management API for local users

### ISSUE THREAD

**Issue by pbrezina** [pbrezina/sssd-test#3237]
Cloned from Pagure issue: https://pagure.io/SSSD/sssd/issue/3242
- Created at 2016-11-21 15:53:59 by [jhrozek](https://pagure.io/user/jhrozek)
- Closed at 2020-03-24 14:04:14 as wontfix
- Assigned to nobody
---

The InfoPipe responder should allow its consumer to change attributes of local users coming from id_provider=files.

There are two basic use-cases:
    - let an application manage local users via an API rather than spawn shadow-utils binaries or libuser
    - add extended attributes for local users. This use-case might be split into its own ticket, but the basic idea is to add attributes keyed by UID into (a separate to keep admins from deleting) database.

### Comments

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-21 15:54:14](https://pagure.io/SSSD/sssd/issue/3242#233449)**

Fields changed

blockedby:  => 2228

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-21 16:34:03](https://pagure.io/SSSD/sssd/issue/3242#233450)**

It should be possible if configure in future to manage remote users too. We should probably fork a separate ticket and consider this use case in the design of this feature.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-23 13:57:49](https://pagure.io/SSSD/sssd/issue/3242#233451)**

We also need to take audit requirements into account when developing this feature. For anyone implementing this in the future, check https://bugzilla.redhat.com/show_bug.cgi?id=1115188 as an example of what was requested for the local provider (which will probably be deprecated in favor of the files provider).

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-25 11:10:52](https://pagure.io/SSSD/sssd/issue/3242#233452)**

Fields changed

milestone: NEEDS_TRIAGE => SSSD Future releases (no date set yet)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2016-11-25 11:13:47](https://pagure.io/SSSD/sssd/issue/3242#233453)**

Fields changed

rhbz:  => todo

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-02-24 14:47:41](https://pagure.io/SSSD/sssd/issue/3242#233454)**

**Metadata Update from @jhrozek**:
- Issue marked as blocked by: #2228
- Issue marked as depending on: #2228
- Issue set to the milestone: SSSD Future releases (no date set yet)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2017-09-05 11:52:08](https://pagure.io/SSSD/sssd/issue/3242#463564)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from 0)
- Custom field mark reset (from 0)
- Custom field patch reset (from 0)
- Custom field review reset (from 0)
- Custom field sensitive reset (from 0)
- Custom field testsupdated reset (from 0)
- Issue close_status updated to: None
- Issue set to the milestone: SSSD 2.0 (was: SSSD Future releases (no date set yet))

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-03-20 13:53:17](https://pagure.io/SSSD/sssd/issue/3242#500563)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue priority set to: minor (was: major)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2018-08-13 11:02:25](https://pagure.io/SSSD/sssd/issue/3242#525879)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue set to the milestone: SSSD 2.1 (was: SSSD 2.0)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2019-02-22 15:52:11](https://pagure.io/SSSD/sssd/issue/3242#555723)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue set to the milestone: SSSD 2.2 (was: SSSD 2.1)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2019-06-13 23:24:04](https://pagure.io/SSSD/sssd/issue/3242#576122)**

**Metadata Update from @jhrozek**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue set to the milestone: SSSD 2.3 (was: SSSD 2.2)

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-11 11:38:53](https://pagure.io/SSSD/sssd/issue/3242#632256)**

**Metadata Update from @thalman**:
- Custom field design_review reset (from false)
- Custom field mark reset (from false)
- Custom field patch reset (from false)
- Custom field review reset (from false)
- Custom field sensitive reset (from false)
- Custom field testsupdated reset (from false)
- Issue tagged with: Canditate to close

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:04:13](https://pagure.io/SSSD/sssd/issue/3242#636445)**

Thank you for taking time to submit this request for SSSD. Unfortunately this issue was not given priority and the team lacks the capacity to work on it at this time.

Given that we are unable to fulfill this request I am closing the issue as wontfix.

If the issue still persist on recent SSSD you can request re-consideration of this decision by reopening this issue. Please provide additional technical details about its importance to you.

Thank you for understanding.

---

**Comment from [jhrozek](https://pagure.io/user/jhrozek) at [2020-03-24 14:04:14](https://pagure.io/SSSD/sssd/issue/3242#636446)**

**Metadata Update from @pbrezina**:
- Issue close_status updated to: wontfix
- Issue status updated to: Closed (was: Open)
